presto-verifier/src/main/java/com/facebook/presto/verifier/QueryRewriter.java

```diff
@@ -108,20 +116,37 @@ public Query shadowQuery(Query query)
     private Query rewriteCreateTableAsSelect(Connection connection, Query query, CreateTableAsSelect statement)
             throws SQLException
     {
+        QualifiedName temporaryTableName = generateTemporaryTableName(statement.getName());
+        Statement rewritten = new CreateTableAsSelect(temporaryTableName, statement.getQuery(), statement.isNotExists(), statement.getProperties(), statement.isWithData());
+        String createTableAsSql = formatSql(rewritten, Optional.empty());
+        String checksumSql = checksumSql(getColumns(connection, statement), temporaryTableName);
+        String dropTableSql = dropTableSql(temporaryTableName);
+        return new Query(query.getCatalog(), query.getSchema(), ImmutableList.of(createTableAsSql), checksumSql, ImmutableList.of(dropTableSql), query.getUsername(), query.getPassword(), query.getSessionProperties());
+    }
+
+    private Query rewriteInsertQuery(Connection connection, Query query, Insert statement)
+            throws SQLException
+    {
+        QualifiedName temporaryTableName = generateTemporaryTableName(statement.getTarget());
+        Statement createTemporaryTable = new CreateTable(temporaryTableName, ImmutableList.of(new LikeClause(statement.getTarget(), Optional.of(INCLUDING))), true, ImmutableMap.of());
+        String createTemporaryTableSql = formatSql(createTemporaryTable, Optional.empty());
+        String insertSql = formatSql(new Insert(temporaryTableName, Optional.empty(), statement.getQuery()), Optional.empty());
```

Comment:
You need to include the columns list from the original query. Otherwise you could have something like:
```
A:
int x
int y

INSERT INTO a (y, x) SELECT 1 y, 2 x;
```

that would be rewritten to `INSERT INTO a SELECT 1 y, 2 x`